ndscan/experiment/fragment.py

```diff
@@ -525,36 +521,45 @@ def bind_param(self, param_name: str, source: ParamHandle) -> Any:
         ``Fluoresce``, binding its intensity and detuning parameters to values and
         defaults appropriate for those particular tasks.
 
+        Transitive binding is supported: if the source parameter is itself bound /
+        overridden, this parameter will be bound to the ultimate source.
+
         See :meth:`override_param`, which sets the parameter to a fixed value/store.
 
         :param param_name: The name of the parameter to be bound (i.e.
             ``self.<param_name>``). Must be a free parameter of this fragment (not
             already bound or overridden).
-        :param source: The parameter to bind to. Must be a free parameter of its
-            respective owner.
+        :param source: The parameter to bind to. Can be free, overridden or rebound.
         """
         param = self._free_params.get(param_name, None)
         assert param is not None, f"Not a free parameter: '{param_name}'"
 
-        # We don't support "transitive" binding for parameters that are already bound.
-        assert source.name in source.owner._free_params, \
-            "Source parameter is not a free parameter; already bound/overridden?"
-
-        own_type = type(self._free_params[param_name])
-        source_type = type(source.owner._free_params[source.name])
-        assert own_type == source_type, (
-            f"Cannot bind {own_type.__name__} '{param_name}' " +
-            f"to source of type {source_type.__name__}")
-
+        # To support "transitive" binding of parameters, follow the chaining of
+        # rebindings until the top-level source is found.
+        toplevel_handle = source._get_toplevel_handle()
```

Comment:
Do we need some check here to avoid a cycle being created (or rather, since we do get the top-level handle always, a parameter being bound to itself)? Maybe an assertion that `toplevel_handle is not param_handle` would improve the user-facing error messages here? (This + tests can wait until a follow-up PR.)